### DISCUSSION TITLE
Watchers Rules

### DISCUSSION THREAD

**Discussion by montgomery102** [TheNickOfTime/handbrake-web#253] [Q&A]
Greetings! Can you explain some of the Watcher Rule requirements? see some examples below:

Rule to look for file extensions, should the extension be written as .mkv or mkv?
Rule to look for file size, is this in GB, MB, or KB? 

Any other useful explanations that I'm not asking for would be greatly appreciated. I'm also willing to submit a pull request to update the readme.md with the information from discussion#229 and this one if that would help you at all. 

Thank you for the awesome project!

**Comment by TheNickOfTime** [comment]
Hi, sorry for the lack of documentation on this topic (and everything else lol), just haven't had the bandwidth to get to it 😭. In the meantime, if you are so inclined to write up documentation, I'd enthusiastically welcome any PRs for it! It currently doesn't exist, but I plan on keeping wiki related stuff at the path `/docs` in the repo and using a workflow to copy it to the repo's wiki. Hopefully by the time you read this, I'll have set up that folder.

But to answer your questions:
For extensions, it should be the full extension with a period - so `.mkv` or `.mp4`. Simply using `mkv` or `mp4` will not match currently. That said, supporting either option would probably be a good improvement I can consider for the future.

For anything size related, it will be in MB. I forgot to put it's metric next to it in the dropdown - I believe I have already fixed this in the v0.8 branch.

Progress on v0.8 is slow right now, but documenting and adding additional watcher options is something that is planned. Right now you can probably accomplish some cool other rules using the regex matching option - though these can be pretty complex if you aren't familiar how they work.
